plugin-server/src/utils/recorded-fetch.ts

```diff
@@ -0,0 +1,188 @@
+import { Headers, RequestInfo, RequestInit, Response } from 'node-fetch'
+
+import { defaultConfig } from '../config/config'
+import { trackedFetch } from './fetch'
+import { UUIDT } from './utils'
+
+export interface RecordedRequest {
+    url: string
+    method: string
+    headers: Record<string, string>
+    body?: string | null
+    timestamp: Date
+}
+
+export interface RecordedResponse {
+    status: number
+    statusText: string
+    headers: Record<string, string>
+    body: string | null
+    timestamp: Date
+}
+
+export interface RecordedHttpCall {
+    id: string
+    request: RecordedRequest
+    response: RecordedResponse
+    error?: Error
+}
+
+export class HttpCallRecorder {
+    private calls: RecordedHttpCall[] = []
+
+    public getCalls(): RecordedHttpCall[] {
+        return [...this.calls]
+    }
+
+    public clearCalls(): void {
+        this.calls = []
+    }
+
+    // This method is needed internally by recordedFetch
+    public addCall(call: RecordedHttpCall): void {
+        this.calls.push(call)
+    }
+}
+
+// Global recorder instance
+export const globalHttpCallRecorder = new HttpCallRecorder()
+
+/**
+ * A wrapper around trackedFetch that conditionally records HTTP requests and responses
+ * based on config flags. If recording is disabled, it simply passes through to trackedFetch.
+ */
+export async function recordedFetch(
+    recorder: HttpCallRecorder,
```

Comment:
I'm confused. This takes a recorder as the first arg but on the imports it is loading it from the global thing.... Why does it need the recorder as an arg?